app/main/views.py

```diff
@@ -104,21 +104,28 @@ def update(service_id, section):
         list(request.form.items()) + list(request.files.items())
     )
 
-    errors = Validate(content, service, posted_data, s3_uploader).errors
+    form = Validate(content, service, posted_data, s3_uploader)
+    update = {}
 
     for question_id in posted_data:
-        if question_id not in errors:
-            service_loader.set(service, question_id, "new value")
-
-    service_loader.post(service)
+        if question_id not in form.errors and question_id in form.clean_data:
```

Comment:
I was thinking about `validate()`, but I actually quite like the `.errors`. It forces you to deal with errors first before using `cleaned_data`, which is somewhat similar to the Django's `is_valid`.

Reversing the conditions will throw an exception, since clean_data is `None` by default. The same thing would happened if we had the `validate()` method and forgot to call it.